app/models/user.rb

```diff
@@ -8,26 +10,44 @@ class User < ApplicationRecord
     format: {with: VALID_EMAIL_REGEX},
     uniqueness: {case_sensitive: false}
 
-  validates :password, presence: true, length: {minimum: 6}
+  validates :password, presence: true, length: {minimum: 6}, allow_nil: true
```

Comment:
Dòng `update_attributes remember_digest: User.digest(remember_token)` chạy mỗi khi user đăng nhập vào và có tích vào ô remember me ở trang log in.
Dòng này update remember_digest = remember_token sau khi đc hash và thay đổi thẳng vào trong cột remember_digest ở db.
Nhưng để thay đổi data trong db thì phải thay đổi cả password do có sự xuất hiện của hàm has_secure_password ở trên nên nếu chỉ đơn thuần là tích vào remember me khi đăng nhập dòng update_attributes sẽ bị bỏ qua. Dẫn đến việc khi tắt browser và bật lại user sẽ không được lưu lại.
Vì thế em mới thêm `allow_nil: true` để không cần thay đổi password mà dòng update_attributes vẫn được chạy